packages/millicast-multiview-demo/multi.js

```diff
@@ -20,68 +24,196 @@ viewer.on('broadcastEvent', (event) => {
+    mainVideoContainer.removeChild(video)
+  } else {
+    remoteVideosContainer.removeChild(video)
+  }
+  delete sourcesTracks[sourceId]
+  delete transceiverToSourceIdMap[videoMediaId]
+}
+
+const sourcesDropDown = document.getElementById('sourcesDropDown')
+const layersDropDown = document.getElementById('layersDropDown')
+
+const updateLayers = (layers) => {
+  const sourceId = sourcesDropDown.value === 'main' ? null : sourcesDropDown.value
+
+  const videoMediaId = sourcesTracks[sourceId]?.find(track => track.media === 'video').mediaId || null
+  const activeLayers = layers[videoMediaId]?.active || []
+  
+  const selectedLayer = layersDropDown.value
+
+  layersDropDown.innerHTML = `<option hidden selected>Select a source</option>` + activeLayers.map(layer => {
+    return `<option value="${layer.id}" ${layer.id === selectedLayer ? 'selected' : ''}>${layer.width}p</option>`
+  }).join('')
 }
 
-const createVideoElement = (mediaStream, sourceId) => {
+const createVideoElement = (mediaStream, videoMediaId) => {
   const video = document.createElement('video')
-  // remoteVideos is already created in the HTML
-  const remoteVideos = document.getElementById('remoteVideos')
 
-  video.id = sourceId || 'main'
+  video.id = videoMediaId 
   video.srcObject = mediaStream
   video.autoplay = true
   // We mute the video so autoplay always work, this can be removed (https://developer.chrome.com/blog/autoplay/#new-behaviors)
   video.muted = true
+  
+  if (videoMediaId === mainTransceiver) {
+    mainVideoContainer.appendChild(video)
+  } else {
+    remoteVideosContainer.appendChild(video)
+  }
+}
+
+const createVideoEventListener = (mediaId) => {
+  const video = document.getElementById(mediaId)
+  video.addEventListener('click', () => { 
+    const selectedSourceId = transceiverToSourceIdMap[mediaId]
+    const mainSoruceId = transceiverToSourceIdMap[mainTransceiver]
+
+    viewer.project(mainSoruceId, sourcesTracks[selectedSourceId])
+
+    viewer.project(selectedSourceId, sourcesTracks[mainSoruceId])
 
-  remoteVideos.appendChild(video)
+    sourcesTracks[selectedSourceId].find(track => track.trackId === 'video').mediaId = mainTransceiver
+    sourcesTracks[mainSoruceId].find(track => track.trackId === 'video').mediaId = mediaId
+    
+    transceiverToSourceIdMap[mainTransceiver] = selectedSourceId
+    transceiverToSourceIdMap[mediaId] = mainSoruceId
+  })
 }
+
+const addSourceOption = (sourceId) => {
+  if (sourceId === 'main') {
+    sourcesDropDown.value = sourceId
+  }
+  const option = document.createElement('option')
+  option.value = sourceId
+  option.text = sourceId
+  sourcesDropDown.add(option)
+}
+
+const removeSourceOption = (sourceId) => {
+  const sourceToRemove = sourcesDropDown.querySelector(`option[value="${sourceId}"]`)
+  if (sourceToRemove) {
+    sourcesDropDown.removeChild(sourceToRemove)
+  }
+}
+
+sourcesDropDown.addEventListener('change', () => {
+  layersDropDown.innerHTML = ''
```

Comment:
It would be nice to have if we could store all the layer info. This causes the layers dropdown to stay empty until a new layers event arrives.